tools/hlint.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+usage() { echo "Usage: $0 -f [all, changeset] -m [check, inplace]" 1>&2; exit 1; }
+
+files=''
+check=true
+
+while getopts ':f:m:' opt
+ do
+     case $opt in
+         f) f=${OPTARG}
+            if [ "$f" = "all" ]; then
+              files=$(find libs/ services/ -not -path "*/test/*" | grep \.hs\$)
+            elif [ "$f" = "changeset" ]; then
+              files=$(git diff --name-only HEAD~ | grep \.hs\$)
+            else
+              usage
+            fi
+            ;;
+         m) m=${OPTARG}
+            if [ "$m" = "inplace" ]; then
+              check=false
+            elif [ "$m" = "check" ]; then
+              check=true
+            else
+              usage
+            fi
+            ;;
+         *) usage;;
+     esac
+done
+
+if [ -z "${f}" ] || [ -z "${m}" ]; then
+    usage
+fi
+
+count=$(echo "$files" | wc -l | xargs echo)
+
+echo "Analysing $count file(s)…"
+
+for f in $files
```

Comment:
For filenames containing SP, TAB, or NL (at least), this will iterate multiple times for a single file.

Haskell modules can't have any "funky" characters in them, so that should be fine to commit/merge; just an FYI.